src/rateEngine/LoadProfileScaler.ts

```diff
@@ -24,58 +34,69 @@ class LoadProfileScaler {
     return this.to(scaler);
   }
 
-  toAverageMonthlyBill(
-    amount: number,
-    rate: RateInterface
-  ): LoadProfile {
-    // Need this as a percent for better resolution within the algorithm
-    const scalerAsPercent = 100;
-    const fnParams = [scalerAsPercent, rate, this];
+  toAverageMonthlyBill(amount: number, rate: RateInterface, goalSeekParams: GoalSeekArgs = {}): LoadProfile {
+    const magnitude = Math.max(Math.floor(Math.log10(Math.abs(amount))), 0);
+
+    const magnitudeScaler = 10 ** magnitude;
```

Comment:
this is the meat of the change... rather than using a set starting point that has 3 digits, we scale the value up to the number of zeros we have in the target amount. this allows the algorithm to adjust the scaling factor according to the "error" left in the result without wildly swinging the next resulting iteration.